### PR TITLE
Migrate custom prescription entry to Mobius

### PR DESCRIPTION
**Pull request by sanchita**
https://www.pivotaltracker.com/story/show/172531165

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 1]
```diff
@@ -28,13 +28,17 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
       model: CustomPrescriptionEntryModel,
       prescription: PrescribedDrug
   ): Next<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
-    var updatedModel = model
-        .drugNameChanged(prescription.name)
+    return if (prescription.isDeleted) {
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt:30]: Ugh, it took me a moment to realise how this works because the deletion is happening elsewhere and we are depending on a reactive change notification to get this.

Ideally, I'd like to change this to make the deletion of the prescribed drug explicit, i.e., forward an event from the UI to the loop once the deletion is done and change loading of prescribed drug to an synchronous call instead of a reactive one.

What do you think? If this sounds good, we can add a comment for now explaining how this works and then add a ticket for fixing this flow.

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryControllerTest.kt [hunk 2]
```diff
@@ -263,19 +262,23 @@ class CustomPrescriptionEntryControllerTest {
...
+    instantiateFixture(openAsNew)
+  }
 
-    uiEvents
-        .compose(controller)
-        .subscribe { uiChange -> uiChange(ui) }
+  private fun createSheetAsUpdate(prescriptionUuid: UUID) {
+    val openAsUpdate = OpenAs.Update(prescriptionUuid)
+    instantiateFixture(openAsUpdate)
+  }
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryControllerTest.kt:270]: Nice! Can we rename these methods to be more use-case focused?

Ex: `createSheetForRecordingNewPrescription` and `createSheetForUpdatingPrescription`.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt [hunk 3]
```diff
@@ -98,7 +98,7 @@ class CustomPrescriptionEntryEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .observeOn(io)
-          .map { prescriptionRepository.prescriptionImmediate(it.prescriptionUuid) }
+          .map { prescriptionRepository.prescription(it.prescriptionUuid) }
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: This will break the remove prescription flow since we are implicitly depending on the reactive change notification to detect if a prescription has been deleted.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 4]
```diff
@@ -25,19 +25,12 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
...
       dispatch(CloseSheet)
     } else {
-      var updatedModel = model
-          .drugNameChanged(prescription.name)
-
-      if (prescription.dosage != null)
-        updatedModel = updatedModel.dosageChanged(prescription.dosage)
-
-      next(updatedModel, SetMedicineName(prescription.name), SetDosage(prescription.dosage))
+      dispatch(SetMedicineName(prescription.name), SetDosage(prescription.dosage))
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: Not necessary to fix this, but this change is unrelated to the commit message. Was it meant to be a separate commit?

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt [hunk 5]
```diff
@@ -32,10 +34,29 @@ class CustomPrescriptionEntryEffectHandler @AssistedInject constructor(
...
+    return ObservableTransformer { effects ->
+
+      val prescription = effects
+          .map { prescriptionRepository.prescriptionImmediate(it.prescriptionUuid) }
+          .filterAndUnwrapJust()
+
+      effects
+          .observeOn(io)
+          .withLatestFrom(prescription)
+          .flatMap { (effect, prescription) ->
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: I saw that this was fixed in a later commit.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 6]
```diff
@@ -15,10 +16,23 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
...
+      model: CustomPrescriptionEntryModel,
+      prescription: PrescribedDrug
+  ): Next<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
+    var updatedModel = model
+        .drugNameChanged(prescription.name)
+
+    if (prescription.dosage != null)
+      updatedModel = updatedModel.dosageChanged(prescription.dosage)
+
+    return next(updatedModel, SetMedicineName(prescription.name), SetDosage(prescription.dosage))
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: Saw that this was changed in a later commit.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt [hunk 7]
```diff
@@ -8,9 +8,28 @@ class CustomPrescriptionEntryUiRenderer(val ui: CustomPrescriptionEntryUi) : Vie
...
+  }
+
+  /**
+   * The dosage field shows a default text as "mg". When it is focused, the cursor will
+   * by default be moved to the end. This will force the user to either move the cursor
+   * to the end manually or delete everything and essentially making the placeholder
+   * useless. As a workaround, we move the cursor to the starting again.
+   */
+  private fun showDefaultDosagePlaceholder(dosage: String?, dosageHasFocus: Boolean?) {
+    if (dosage == null || dosageHasFocus == null) return
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:30]: This cannot be moved to the top level function as it will break the logic. 
These checks are required to render the dosage placeholder.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt [hunk 8]
```diff
@@ -1,9 +1,16 @@
...
+import org.simple.clinic.util.ValueChangedCallback
 
 class CustomPrescriptionEntryUiRenderer(val ui: CustomPrescriptionEntryUi) : ViewRenderer<CustomPrescriptionEntryModel> {
 
+  private val drugNameValueCallback = ValueChangedCallback<Boolean>()
+
   override fun render(model: CustomPrescriptionEntryModel) {
+    if (model.drugName == null) return
+
+    val isDrugNameNotBlank = model.drugName.isNotBlank()
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt]: Is this necessary?
Since it is not being reused I don't think we need this change.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 9]
```diff
@@ -14,15 +12,15 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
     return when (event) {
       is CustomPrescriptionDrugNameTextChanged -> next(model.drugNameChanged(event.name))
       is CustomPrescriptionDrugDosageTextChanged -> next(model.dosageChanged(event.dosage))
-      SaveCustomPrescriptionClicked -> createNewPrescriptionEntry(model)
+      SaveCustomPrescriptionClicked -> createOrUpdatePrescriptionEntry(model)
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt:17]: I was following naming conventions in other places. Eg: `BloodPressureEntryEffectHandler`.
I can make this change.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt [hunk 10]
```diff
@@ -8,9 +8,28 @@ class CustomPrescriptionEntryUiRenderer(val ui: CustomPrescriptionEntryUi) : Vie
...
+   * useless. As a workaround, we move the cursor to the starting again.
+   */
+  private fun showDefaultDosagePlaceholder(dosage: String?, dosageHasFocus: Boolean?) {
+    if (dosage == null || dosageHasFocus == null) return
+
+    when {
+      dosage.isBlank() && dosageHasFocus -> ui.setDrugDosageText(DOSAGE_PLACEHOLDER)
+      dosage.trim() == DOSAGE_PLACEHOLDER && dosageHasFocus.not() -> ui.setDrugDosageText("")
+      dosage.trim() == DOSAGE_PLACEHOLDER && dosageHasFocus -> ui.moveDrugDosageCursorToBeginning()
+    }
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:36]: Yes. You are right about this. I was planning on doing this refactor in the end but forgot about it. Should have added a `TODO`. 
I will do this now.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt]: Yeah, it's not necessary. I just prefer doing this to hide implement details and expose intent. If you prefer to keep it this way, it's fine too.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt [hunk 11]
```diff
@@ -10,12 +10,16 @@ class CustomPrescriptionEntryUiRenderer(val ui: CustomPrescriptionEntryUi) : Vie
...
-    if (model.drugName == null) return
+    toggleSaveButton(model.drugName)
 
-    val isDrugNameNotBlank = model.drugName.isNotBlank()
-    drugNameValueCallback.pass(isDrugNameNotBlank, ui::setSaveButtonEnabled)
     displaySheetTitle(model.openAs)
+  }
 
+  private fun toggleSaveButton(drugName: String?) {
+    if (drugName == null) return
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:17]: This is part of the logic of toggling the save button. If I move this to top level function, the tests will break and the functionality as well. Reason being that it is possible to enter the dosage first and then the drug name. So it is perfectly acceptable to have the `drugName` as `null` but for enabling save button, this has to be non-null.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 12]
```diff
@@ -15,10 +16,23 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
       is CustomPrescriptionDrugDosageFocusChanged -> next(model.dosageFocusChanged(event.hasFocus))
       SaveCustomPrescriptionClicked -> createOrUpdatePrescriptionEntry(model)
       CustomPrescriptionSaved -> dispatch(CloseSheet)
-      is CustomPrescriptionFetched -> TODO()
+      is CustomPrescriptionFetched -> onPrescriptionFetched(model, event.prescription)
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: I am following conventions elsewhere. I can change this.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt [hunk 13]
```diff
@@ -1,3 +1,7 @@
 package org.simple.clinic.drugs.selection.entry
 
+import java.util.UUID
+
 sealed class CustomPrescriptionEntryEffect
+
+data class SaveCustomPrescription(val patientUuid: UUID, val drugName: String, val dosage: String?) : CustomPrescriptionEntryEffect()
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt:7]: iirc, `Optional` is for cases where are using the value in `Rx` streams and `Nullable` is for immediate use.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt [hunk 14]
```diff
@@ -18,6 +31,44 @@ class CustomPrescriptionEntryEffectHandler @AssistedInject constructor(
...
+          .observeOn(io)
+          .map {
+            val user = userSession.loggedInUserImmediate()!!
+            val currentFacility = facilityRepository.currentFacilityImmediate(user)!!
+            SavePrescription(
+                patientUuid = it.patientUuid,
+                name = it.drugName,
+                dosage = it.dosage,
+                facility = currentFacility
+            )
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: This has been refactored in the later commits to extract creating of the `SavePrescription` object.
Does it make sense to not do this anymore?

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 15]
```diff
@@ -10,6 +10,9 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
       Next<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
     return when (event) {
       is CustomPrescriptionDrugNameTextChanged -> next(model.drugNameChanged(event.name))
+      is CustomPrescriptionDrugDosageTextChanged -> next(model.dosageChanged(event.dosage))
+      SaveCustomPrescriptionClicked -> TODO()
+      CustomPrescriptionSaved -> TODO()
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: 😞

File: app/src/main/java/org/simple/clinic/drugs/PrescriptionRepository.kt [hunk 16]
```diff
@@ -127,8 +129,11 @@ class PrescriptionRepository @Inject constructor(
         .toObservable()
   }
 
+  @Deprecated("Use prescriptionImmediate() instead")
   fun prescription(prescriptionUuid: UUID): Observable<PrescribedDrug> = dao.prescription(prescriptionUuid).toObservable()
 
+  fun prescriptionImmediate(prescriptionUuid: UUID): Optional<PrescribedDrug> = dao.prescriptionImmediate(prescriptionUuid).toOptional()
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/PrescriptionRepository.kt]: That's a good point. I will make this return `PrescribedDrug`.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt (hunk 13 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt:7]: Right, but that's because Rx lambdas will throw exceptions if an operator returns `null`. I'm thinking if we need to use the `Optional` type itself to expose some intent here.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt (hunk 14 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: Yeah, I was questioning the need for the `SavePrescription` object itself since it doesn't actually anything. The facility can be queried in line when making the `savePrescription` call.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 1 above)
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt:30]: I will raise a ticket for this.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 4 above)
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: This was breaking the tests. I realized that this change was not necessary although I was confused about it for some time. We would display the fetched prescription in the UI but if we do not update the model to reflect the current value of `drugName` and `dosage`, will that be an unexpected UI state?
Rn, we only update the `drugName` and `dosage` when there is `TextChanged` callback.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 9 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt:17]: Yeah, it's up to you.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt (hunk 10 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:36]: This feature has broken as part of the refactor, so if moving this out is simpler and fixes the problem, we can do it now.

Otherwise, I am fine with fixing the problem as is and raising a ticket to do this later since it's already a large-ish PR.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt (hunk 11 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:17]: Make sense.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 12 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: It's up to you. No strong feelings either way.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 4 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: Ah, alright.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt (hunk 3 above)
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: Fixed this by retaining and using the reactive version of this method.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryInit.kt [hunk 17]
```diff
@@ -3,10 +3,14 @@ package org.simple.clinic.drugs.selection.entry
...
 import com.spotify.mobius.Init
+import org.simple.clinic.mobius.first
 
 class CustomPrescriptionEntryInit : Init<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
 
   override fun init(model: CustomPrescriptionEntryModel): First<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
-    return first(model)
+    return when (model.openAs) {
+      is OpenAs.New -> first(model)
+      is OpenAs.Update -> first(model, FetchPrescription(model.openAs.prescribedDrugUuid))
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryInit.kt:13]: Fixed

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt [hunk 18]
```diff
@@ -15,10 +16,23 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
...
 
+  private fun onPrescriptionFetched(
+      model: CustomPrescriptionEntryModel,
+      prescription: PrescribedDrug
+  ): Next<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
+    var updatedModel = model
+        .drugNameChanged(prescription.name)
+
+    if (prescription.dosage != null)
+      updatedModel = updatedModel.dosageChanged(prescription.dosage)
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: This code has been simplified and removed in the future commits. I will not be making this change then

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt (hunk 10 above)
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt:36]: I will have to make this change in order to make it work. But it will be difficult to change this commit now.
I will see if I can drop this commit and create a new commit and reorder it.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt (hunk 14 above)
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandler.kt]: Oh I got what you mean. I will make this change.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 15 above)
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: This error is not reported by running tests. Whereas migration to Mobius is dependant on tests largely. So do you think we should update the doc to add a step where after we move each method, we should run the app in the emulator as well?

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryModel.kt [hunk 19]
```diff
@@ -7,12 +7,13 @@ import kotlinx.android.parcel.Parcelize
 data class CustomPrescriptionEntryModel(
     val openAs: OpenAs,
     val drugName: String?,
-    val dosage: String?
+    val dosage: String?,
+    val dosageHasFocus: Boolean?
```
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryModel.kt:11]: Currently, we need this to be a nullable boolean in order to distinguish between when this value is explicitly vs implcitly set. This makes the dosage placeholder logic work and the tests pass.
I don't want to change the tests while migrating this screen.
A better solution, like you said, would be to move this presentation logic out of Mobius loop.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt (hunk 13 above)
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt:7]: Since `dosage` in the model is also nullable and not `Optional`, I'd much rather keep it this way for now.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt (hunk 10 above)
[the remark this answers is not in the document]
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt]: I ended up fixing the commit.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt (hunk 13 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt:7]: Yeah, this was just a wandering thought.

File: app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt (hunk 15 above)
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt]: This should ideally not happen and the test should have broken. Any thoughts on why it did not break?
